android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -258,6 +260,23 @@ fun invalidatePurchaserInfoCache() {
     Purchases.sharedInstance.invalidatePurchaserInfoCache()
 }
 
+fun canMakePayments(context: Context,
+                    features: List<String>,
+                    onResult: OnResultAny<Boolean>) {
```

Comment:
🤔 
This would break when we add Amazon, right? 
I don't see much of a way around that, tbh, at least until the Amazon stuff gets merged. 

On another note, I feel like this might lead to weird usage from a customer's perspective: 
For example, they might be using react-native and have to call `Purchases.canMakePayments("SUBSCRIPTIONS")`, while making sure that they're passing in names that are compatible with Android's `BillingFeature`, right? 

Perhaps we can create an abstraction of that? Either by deciding which features we want to check ourselves, or by having an enum or something abstracting the Google Billing types? That way React-Native users on iOS won't have to look up Android's `BillingFeatures` to call, and it also gives us a clean way of separating Amazon vs Google Play features?

Reply:
@aboedo so i do have a BillingFeature enum on the ReactNative side, and was planning to do that for each hybrid. So the call for an end user would use an enum type. And then I have the BillingFeature enum in `purchases-android`, which is the one I mapped to here.

Re-declaring our purchases enums seemed to be the norm for other enums in ReactNative, I assume because enum types aren't compatible across all the languages? We do that for PURCHASE_TYPE, for instance, and pass a string to  `common` methods expecting it, and then the mapping back to the native enum happens there. 

So there really shouldn't be a reason that the InvalidArgumentException is thrown, since the interfaces in each hybrid will enforce the correct mappings. Let me know if I just missed something? I can open the ReactNative PR for reference, too 

Could you elaborate on why this would break Amazon?

Reply:
That makes sense. 

And yeah, enums wouldn't really work here, don't know what I was thinking. 

> Could you elaborate on why this would break Amazon?

🤔 I should clarify: I don't think this would break Amazon, rather, we'd have to add logic to make a different call for Amazon, since `BillingFeature` is a Google Play thing, right? I'm not sure Amazon even provides a method to check if purchases are allowed on the device. 

@vegaro might know more about that part, but we probably won't have to worry about it until Amazon is ready to ship

Reply:
ahhh i see. maybe we could have a check for the store and just return an error if amazon. let me know your thoughts @vegaro

Reply:
Good catch @aboedo. It's tricky. What we would normally do is `canMakePayments` would be an abstract function in [BillingAbstract](https://github.com/RevenueCat/purchases-android/blob/develop/common/src/main/java/com/revenuecat/purchases/common/BillingAbstract.kt) and then the Amazon module would have an implementation different to the one Google one.

Since `canMakePayments` lives in Purchases and it's actually a static call, it doesn't use an instance of BillingClient, it can't be a static abstract function. I also don't think Amazon provides an equivalent to `canMakePayments` 🤔 Maybe we could check if the package of the store is present? Probably error prone though...

Reply:
Re-reading this. @aboedo I don't think this code in `purchases-hybrid-common` would break in Amazon, since `BillingFeature` is an enum in `purchases-android`. The call to `Purchases.Companion.canMakePayments` will break though, because its implementation assumes that the BillingClient is  able to connect to play services

Reply:
@aboedo I discussed this with Cesar. Proposed plan is:
* add a new overload of this function that takes a store parameter
* default that store parameter to Play for now, and return false if it's not Play
* once that's in, i'll update purchases-hybrid-common and hybrid repos off the amazon branches

does that sound ok to you?

Reply:
Sounds great! 👌